tests/TopDownProteomics.Tests/ProteoformClassifierTest.cs

```diff
@@ -0,0 +1,59 @@
+﻿using NUnit.Framework;
+using System.Collections.Generic;
+using System.Linq;
+using TopDownProteomics.ProForma;
+using TopDownProteomics.Proteomics;
+
+namespace TopDownProteomics.Tests
+{
+    [TestFixture]
+    public static class ProteoformClassifierTest
+    {
+        [Test]
+        [TestCase("EMEVEESPEK", 1, "1")]
+        [TestCase("EM[UNIMOD:35]EVEES[UNIMOD:21]PEK", 1, "1")]
+        [TestCase("[+42]-EMEVEES[UNIMOD:21]PEK", 1, "2B")]
+        [TestCase("EM[UNIMOD:35]EVEESPEK-[+42]", 1, "2B")]
+        [TestCase("{Phospho}EM[UNIMOD:35]EVEESPEK", 1, "2A")]
+        [TestCase("{+80}EM[UNIMOD:35]EVEESPEK", 1, "3")]
+        [TestCase("SEQUEN[Formula:C12H20O2]CE", 1, "1")]
+        [TestCase("[Phospho]?EMEVTSESPEK", 1, "2A")]
+        [TestCase("EMEVT[#g1]S[#g1]ES[Phospho#g1]PEK", 1, "2A")]
+        [TestCase("EM[+15.9949]EVEES[-79.9663]PEK", 1, "2B")]
+        [TestCase("EMEVEESPEK[Acetyl|Trimethyl]", 1, "2B")] // RTF: Ask into this!
+        [TestCase("EMEVEE(?SP)EK", 1, "2C")]
+        [TestCase("PROTEOSFORMSISK(?N)", 1, "2C")]
+        [TestCase("EMEVEESPEKB", 1, "2C")]
+        [TestCase("EMEVEESPEKJ", 1, "2C")]
+        [TestCase("EMEVEESPEKX", 1, "2C")]
+        [TestCase("EMEVEESPEKZ", 1, "2C")]
+        [TestCase("EMEVEESPEK", 2, "2D")]
+        [TestCase("PROT(EOSFORMS)[+19.0523]ISKN", 1, "3")]
+        [TestCase("PROT(EOSFORMS)[+19.0523]ISK(?N)", 1, "4")]
+        [TestCase("PROT(EOSFORMS)[+19.0523]ISK(?N)", 2, "5")]
+        [TestCase("PROT(?EOSFORMS)[Oxidation]ISKN", 2, "4")]
+        [TestCase("PROT(?EOSFORMS)[+19.0523]ISKN", 2, "5")]
+        [TestCase("PROT(?EOSFORMS[+19.0523])ISKN", 2, "5")] // RTF: Ask into this! .. incorrect syntax, but should be handled
```

Comment:
I agree this should return `5`

Reply:
agreed and fixed. Also added another test case where the PTM is outside of the ambiguous range -> 4.